backend/routeHandler.js

```diff
@@ -74,6 +74,50 @@ module.exports = function configureRoutes(options) {
     res.send(JSON.stringify(query));
   });
 
+  options.router.post('/getQueryByName', async (req, res) => {
+    try {
+    // Checking templates
+      const getQueryParams = {
+        sourceKey: req.body.sourceKey,
+        type: "template"
+      };
+
+      let response = (await options.getRestClient(req).graphQuery.getQueries(getQueryParams));
+
+      let queries = response.body.filter((q) => {
+        return q.name.toLowerCase() === req.body.name;
+      });
+
+      // Checking statics
+      getQueryParams.type = "static"
+
+      response = (await options.getRestClient(req).graphQuery.getQueries(getQueryParams));
+
+      queries = queries.concat(response.body.filter((q) => {
+        return q.name === req.body.name;
+      }));
+
+      if (queries.length === 0) {
+        throw new Error(`Query ${req.body.name} either does not exist or you do not have access to it.`);
+      }
+      else if (queries.length === 1) {
+        response.body = queries[0]
+        res.status(200);
+        res.contentType('application/json');
+        res.send(JSON.stringify(response));
```

Comment:
Reusing the response object is not straightforward as it could be the response from the queries while you are picking a query template? Are you sure you don't have any impact on it?

I would rather send a dedicated new object (with all the extra other properties if needed):
res.send(JSON.stringify({ body: queries[0] }));

Reply:
fixed: no other parameters are necessary